### ISSUE TITLE
[Bug]: Relative path's on splat routes are broken

### ISSUE THREAD

**Issue by gijsroge** [remix-run/react-router#11038]
### What version of React Router are you using?

6.19.0

### Steps to Reproduce

Create a route with a relative="path" option in a splat route that either goes forward, or backward. Non splat routes are working fine.

### Expected Behavior

The navigation should respect the relative path

### Actual Behavior

It goes back to the root.

Let's say you are on page `/a/b/c` and you link to `d` using `relative="path"` it goes to `/d` instead of `/a/b/c/d`

Reproduction in Remix: https://stackblitz.com/edit/remix-run-remix-jb8sq2?file=app%2Froutes%2F%24.tsx

I'm guessing it has something to do with the [latest change](https://github.com/remix-run/react-router/pull/11006), because if I revert back to an earlier version of Remix there is no issue.

**Comment by kiliman** [comment]
Yes, I think it's wrong as well. When you select `relative="path"`, it should do path resolution against the current **URL**, not the current route (since the default for navigation is relative to _route_ definition). 

For splat routes at the root, the route path would be `/`, not `/a/b/c` as the current URL.

Also, I would say that `/a/b/c` => `d` would be `/a/b/d`. Because technically you would treat `c` as a _file_, not a _folder_.

If you want `/a/b/c/d` then the current URL should be `/a/b/c/` (trailing slash).

```js
let u = new URL('d', 'http://localhost/a/b/c')
// u.pathname = '/a/b/d'
let u = new URL('d', 'http://localhost/a/b/c/')
// u.pathname = '/a/b/c/d'
```

**Comment by gijsroge** [comment]
@kiliman thanks for conforming!

About the trailing slash. Not sure if that's ideal because most websites do not use a trailing slash and this would change the behavior of the `relative="path"` feature and introduce a breaking change.

**Comment by MaximeBernard** [comment]
I can confirm as well. All our apps are "broken" (active tab is not displaying active anymore).

This is a breaking change and should then be flagged as a major update.

**Comment by brophdawg11** [comment]
This looks like a bug related to both #10983 and #11006 - we'll get a fix in for this 👍.  I think we just need the change from #10983 in a second code path.

> if I revert back to an earlier version of Remix there is no issue.

Interestingly enough, this _was_ an issue in prior versions without `relative="path"` (i.e., `<Link to="test">`), but the above PRs caused the issue to show up in `2.3.0` with and without `relative="path"`

> should do path resolution against the current URL

This is not quite accurate.  [`relative=path`](https://reactrouter.com/en/main/components/link#relative) is only intended to impact the resolution of `".."` - does that mean "go up one route in the hierarchy" or "go up one URL segment".  Routes are still relative to the current location in context - which is the bug being fixed by #11006.

**Comment by brophdawg11** [comment]
This is fixed in #11045 and will be available in the next release
